xlators/cluster/dht/src/dht-rebalance.c

```diff
@@ -2359,17 +2360,32 @@ rebalance_task_completion(int op_ret, call_frame_t *sync_frame, void *data)
         op_ret = -1;
     }
 
-    DHT_STACK_UNWIND(setxattr, sync_frame, op_ret, op_errno, NULL);
+    DHT_STACK_UNWIND(setxattr, setxattr_frame, op_ret, op_errno, NULL);
+    GF_ASSERT(syncop_frame->local == NULL);
+    STACK_DESTROY(syncop_frame->root);
     return 0;
 }
 
 int
 dht_start_rebalance_task(xlator_t *this, call_frame_t *frame)
 {
     int ret = -1;
+    call_frame_t *syncop_frame = NULL;
+
+    syncop_frame = copy_frame(frame);
```

Comment:
Why is needed this copy ? the `frame` will exist until `DHT_STACK_UNWIND()` is called inside `rebalance_task_completion()`

Reply:
@xhernandez Thought it is not a good idea to modify pid/lk-owner of frame that is coming from parent xlators. So had to go for copy_frame(). Do let me know if there is a way out. I couldn't think of any.

Reply:
@xhernandez One thing we could probably do is to introduce another api into syncop which takes lk-owner and pid maybe others as arguments to set them to the op-frame. That is one option. Do you think that can be better?

Reply:
I found another place where we can prevent a frame creation. Will update that patch.

Reply:
@xhernandez Added a new commit.

Reply:
I think you are right. A new frame in this case seems ok.